exec/java-exec/src/main/java/org/apache/drill/exec/planner/FileSystemPartitionDescriptor.java

```diff
@@ -157,7 +159,20 @@ private String getBaseTableLocation() {
 
   @Override
   protected void createPartitionSublists() {
-    List<String> fileLocations = ((FormatSelection) table.getSelection()).getAsFiles();
+    Collection<String> fileLocations = null;
+    if (scanRel instanceof DrillScanRel) {
```

Comment:
This seems like it is leaking specific implementations. Can we come up with a way to move this around. Possibly GroupScan should have hasFiles() and getFiles()?